### DISCUSSION TITLE
尝试用宏注册插槽，没有响应

### DISCUSSION THREAD

**Discussion by jinyuejin** [CefView/QCefView#268] [Q&A] [answered]
![image](https://github.com/CefView/QCefView/assets/37995023/1035d5b6-c0e6-4db5-b496-ab5636522e68)
看到官网有这种写法，就试了一下。第一种写法可以触发onQCefQueryRequest；但是第二种就没有响应。
想问一下有解决办法么？

**Comment by L-Super** [comment]
为什么要用Qt4的写法呢

**Reply by jinyuejin** [reply]
因为想实现延迟加载qcefview.dll链接库，用第一种写法会报错:
`LINK : fatal error LNK1194: 无法延迟加载“QCefView.dll”，原因在于数据符号“"__declspec(dllimport) public: static struct QMetaObject const QCefView::staticMetaObject" (__imp_?staticMetaObject@QCefView@@2UQMetaObject@@B)”的导入；链接时不使用 /DELAYLOAD:QCefView.dll`

**Reply by jinyuejin** [reply]
因为 cef 的运行文件很多，所以想把所有的配置、资源、子进程等文件都放入一个文件夹里。就是 可执行文件 和 这个目录同级。
看了一下，好像只能通过延迟加载链接库的方法实现。如果有其他的方法，能解决就好了。

**Reply by L-Super** [reply]
> 因为 cef 的运行文件很多，所以想把所有的配置、资源、子进程等文件都放入一个文件夹里。就是 可执行文件 和 这个目录同级。
> 看了一下，好像只能通过延迟加载链接库的方法实现。如果有其他的方法，能解决就好了。

我理解的意思是，你想要实现这样的应用结构
- application
- cef folder
- - cef lib
- - cef resources

如果是linux 系统，可以通过设置rpath实现。

**Reply by jinyuejin** [reply]
如果是 win 呢？

**Reply by tishion** [reply]
connect如果失败，Qt会输出log，看一下错误原因。

**Reply by jinyuejin** [reply]
学到了。错误是：
`qt.core.qobject.connect: QObject::connect: No such slot QCefView::OnQCefQueryRequest(int, qint64, const cefQueryRequest&) `
看来这种方式，会从父类中寻找。我再调整一下。谢谢！@tishion @L-Super

**Reply by tishion** [reply]
你应该检查是不是函数签名写错了，比如signal/slot的参数类型是否写错了

**Reply by jinyuejin** [reply, marked as answer]
@tishion  实在是抱歉，不知道为什么，复制错了，错误是：
`qt.core.qobject.connect: QObject::connect: No such slot QCefView::OnQCefQueryRequest(int, qint64, const QCefQuery&) in ...文件路径`
错误不是参数的原因。

重新看了官方[示例](https://cefview.github.io/QCefView/docs/guide/CefviewQuery#add-c-code)，使用的是
`connect(cefViewWidget, SIGNAL(cefQueryRequest(int, int, const QCefQuery&)), this, SLOT(onQCefQueryRequest(int, int, const QCefQuery&)));` 
`onQCefQueryRequest`是**MainWindow**的成员函数，而我的`onQCefQueryRequest`是qcefview的子类的成员函数。
